### DISCUSSION TITLE
NextJS / Vercel is fast and slow

### DISCUSSION THREAD

**Discussion by sroussey** [vercel/next.js#45245] [Show and tell]
I converted a hobby site from using pages to the new app directory.

Overall, I like the new setup better. The public pages, which should be almost all server rendered seem fast. The logged in area is more sensitive to network speeds as it uses an api model (will look at changing that!). Which brings me to Vercel...

<img width="814" alt="image" src="https://user-images.githubusercontent.com/127349/214456675-b00976aa-e79b-4aac-890d-7861b1356148.png">

And looking at a single request:

<img width="814" alt="image" src="https://user-images.githubusercontent.com/127349/214456974-4b210b8f-1cc6-4df9-a4ee-b5f177fe7151.png">

Seems like it is Vercel that is slow. 1s waiting for a request to start sending kind of slow.

I am on a hobby tier, and the site is a hobby project, so if it is a pricing thing, then that is fine (should mention on pricing page). Maybe the CDN needs priming, and since the site gets no traffic, I am always hitting new urls as far as they are concerned. I hard-reload a few times, but the timings stick between 500ms and 850ms for the most part. Interestingly, this is not the case for HTML files (< 100ms always), so I feel like some prioritization is going on. I could put cloudflare in front, and see how that goes, but I am curious to optimize first. 

BTW: I am the one that added the Server Timings stuff on the bottom of the second image when hovering on the waterfall in Google Chrome DevTools. Not much use here as the files are all static!

The site is https://embarc.com.

I wrote it up quickly a year or two ago. The data is a year old as I was relying on someone else for that, but going to write something to pull from the SEC myself, and get some better stock quote info, and it should be workable again. It is interesting hobby since that data will only change (get pulled and processed) once a day, making it mostly static site, and a good example. Hopefully, I will figure out some of the speed stuff along the way, so when the data is good again, all will be right with this hobby project.

Then I'll add VCs and smaller startups...

**Comment by sroussey** [comment]
Another thing, using PageSpeed Insights:

https://pagespeed.web.dev/report?url=https%3A%2F%2Fembarc.com%2F

It looks like the pre-fetch stuff is returning text but not using gzip!!

<img width="959" alt="image" src="https://user-images.githubusercontent.com/127349/215582821-b41276b9-fa7a-49e4-8e79-cc46af6a1024.png">

**Reply by digggggggggg** [reply]
Same here, I'm experiencing very inconsistent response times ranging between 100ms and 3000ms. And this is for cached responses, which I imagine should imply it's the fastest possible response. ![image](https://github.com/vercel/next.js/assets/182178/46daede2-64e9-4be4-a780-6124ef9360d5)

**Reply by ZainKhalidOfficial** [reply]
I'm experiencing the same issue.

Any solutions yet?

**Comment by goodallman** [comment]
I am having the same problem. If it's a case of paying for the paid tier, it should be mentioned somewhere, so we can adapt to it.

**Reply by Bashek** [reply]
The same problem, any answer?
